src/components/DeleteComment.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import { deleteCommentByID } from "../api";
+
+export default function DeleteComment({
+  comment_id,
+  setCommentExists,
+  author,
+  user,
+}) {
+  const [loading, setloading] = useState(false);
+  const [err, setErr] = useState(null);
+  const [allow, setAllow] = useState(true);
+
+  const handleDelete = () => {
+    setloading(true);
+    setAllow(true);
+    if (user === author) {
+      deleteCommentByID(comment_id).then((res) => {
+        if (res.status !== 204) {
+          setErr("Something Went Wrong Cant Delete Comment:(");
+        }
+        setCommentExists(false);
+        setloading(false);
+      });
+    }
+    setAllow(false);
+  };
+  if (err) return <h4>{err}</h4>;
+  else if (!allow) return <h4>You can only delete your comments</h4>;
```

Comment:
This message shows up for a second even if I'm the correct user, just before my comment is deleted